### PR TITLE
KAFKA-9404: Use ArrayList instead of LinkedList in Sensor Class

### PR DESCRIPTION
**Pull request by belugabehr**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java [hunk 1]
```diff
@@ -306,7 +305,7 @@ public boolean hasExpired() {
     }
 
     synchronized List<KafkaMetric> metrics() {
```
Comment (rondagostino) [inline clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java:307]: If we are going to change it then we should probably also add a comment for the method saying `// visible for testing`

**Review by rondagostino — COMMENTED** [review]
LGTM

File: clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java (hunk 1 above)
Comment (belugabehr) [reply clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java:307]: Done.  Thank you.  Please review and kick off test.

**Review by rondagostino — COMMENTED** [review]
LGTM

**Comment by ijuma** [comment]
ok to test

**Review by ijuma — APPROVED** [review]
LGTM

File: clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java (hunk 1 above)
Comment (ijuma) [reply clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java:307]: This method is not only used in tests, so the comment seems wrong to me. Can we fix it?

**Review by ijuma — CHANGES_REQUESTED** [review]
Actually, the comment doesn't seem correct. Please remove it.

**Comment by belugabehr** [comment]
@rondagostino @ijuma Requested changes made.

**Review by rondagostino — COMMENTED** [review]
LGTM. Sorry about the mistake on the comment, and thank you for the PR.

**Comment by ijuma** [comment]
ok to test

**Comment by ijuma** [comment]
Single failing test in both jobs, flaky test fixed by:

https://github.com/apache/kafka/commit/f08c9c7de6f7b85bddc6c83c0d852b5b3b1f430a

**Review by ijuma — APPROVED** [review]
LGTM

**Pull request merged**